neural_lam/vis.py

```diff
@@ -67,111 +193,102 @@ def plot_error_map(errors, datastore: BaseRegularGridDatastore, title=None):
 def plot_prediction(
     datastore: BaseRegularGridDatastore,
-    da_prediction: xr.DataArray,
-    da_target: xr.DataArray,
+    da_prediction: xr.DataArray = None,
+    da_target: xr.DataArray = None,
     title=None,
     vrange=None,
+    boundary_alpha=0.7,
+    crop_to_interior=True,
 ):
     """
     Plot example prediction and grond truth.
 
     Each has shape (N_grid,)
 
     """
-    # Get common scale for values
     if vrange is None:
         vmin = min(da_prediction.min(), da_target.min())
         vmax = max(da_prediction.max(), da_target.max())
-    elif vrange is not None:
+    else:
         vmin, vmax = vrange
 
-    extent = datastore.get_xy_extent("state")
-
-    # Set up masking of border region
-    da_mask = datastore.unstack_grid_coords(datastore.boundary_mask)
-    mask_values = np.invert(da_mask.values.astype(bool)).astype(float)
-    pixel_alpha = mask_values.clip(0.7, 1)  # Faded border region
-
     fig, axes = plt.subplots(
         1,
         2,
         figsize=(13, 7),
         subplot_kw={"projection": datastore.coords_projection},
     )
 
-    # Plot pred and target
-    for ax, da in zip(axes, (da_target, da_prediction)):
-        ax.coastlines()  # Add coastline outlines
-        da.plot.imshow(
+    for ax, da, subtitle in zip(
+        axes, (da_target, da_prediction), ("Ground Truth", "Prediction")
+    ):
+        plot_on_axis(
             ax=ax,
-            origin="lower",
-            x="x",
-            extent=extent,
-            alpha=pixel_alpha.T,
+            da=da,
+            datastore=datastore,
             vmin=vmin,
             vmax=vmax,
-            cmap="plasma",
-            transform=datastore.coords_projection,
+            ax_title=subtitle,
+            cmap="viridis",
+            boundary_alpha=boundary_alpha,
+            crop_to_interior=crop_to_interior,
         )
 
-    # Ticks and labels
-    axes[0].set_title("Ground Truth", size=15)
-    axes[1].set_title("Prediction", size=15)
-
     if title:
         fig.suptitle(title, size=20)
 
+    cbar_ax = fig.add_axes([0.2, 0.05, 0.6, 0.03])
+    fig.colorbar(axes[0].collections[0], cax=cbar_ax, orientation="horizontal")
+
     return fig
 
 
 @matplotlib.rc_context(utils.fractional_plot_bundle(1))
 def plot_spatial_error(
-    error, datastore: BaseRegularGridDatastore, title=None, vrange=None
+    error,
+    datastore: BaseRegularGridDatastore,
+    title=None,
+    vrange=None,
+    boundary_alpha=0.7,
+    crop_to_interior=True,
 ):
-    """
-    Plot errors over spatial map
-    Error and obs_mask has shape (N_grid,)
-    """
-    # Get common scale for values
+    """Plot spatial error with projection-aware axes."""
+
+    grid_shape = [datastore.grid_shape_state.x, datastore.grid_shape_state.y]
+
+    value_source = error
+    if hasattr(value_source, "detach"):
+        value_source = value_source.detach()
+    if hasattr(value_source, "cpu"):
+        value_source = value_source.cpu()
```

Comment:
If we assume that this is tensor (and add that as type annotation in the function), we could just always do `.detach().cpu()`.